src/index.js

```diff
@@ -448,14 +453,24 @@ export default class Carousel extends Component {
     if (transition !== 'fade') {
       const leftPos = leftOffset + dragOffset;
       trackStyle = { ...trackStyle,
-        transform: `translateX(${isRTL ? -leftPos : leftPos}px)`,
+        ...isVertical && { transform: `translateY(${isRTL ? -leftPos : leftPos}px)` },
+        ...!isVertical && { transform: `translateX(${isRTL ? -leftPos : leftPos}px)` },
         transition: transitionDuration ? `transform ${ms('' + transitionDuration)}ms ${easing}` : 'none'
       };
     }
+
     if (!draggable) {
       trackStyle.touchAction = 'auto';
     }
+
     const controls = this.getControls();
+    let arrowOffset = 0;
+    if (this._viewport && this._track) {
+      arrowOffset += this._viewport.offsetHeight / 2;
+      arrowOffset -= this.props.verticalArrowPadding;
+    }
+
+    const arrowStyle = { ...isVertical && { ...!arrows && { transform: `translateY(${ arrowOffset }px)` } } };
```

Comment:
Nit: this is a little hard to read, I would prefer something like
```
if (isVertical && arrows) {
  ...
}
```
Also, if this is indeed meant to only apply when `arrows` is false, then I don't really understand these changes as we shouldn't be applying arrow styling if the consumer specifically said that they do not want arrows...

Reply:
I was trying to avoid using `let`.

Reply:
removed this.